### DISCUSSION TITLE
Curl doesn't work properly with 2G and 3G for downloading a small file with HTTP protocol

### DISCUSSION THREAD

**Discussion by khtorabi** [curl/curl#11732] [Q&A]
I have a code with C/C++ in Android(NDK/SDK) to download small files with Curl. I don't have any problem when I am working with 4G and 5G. But when I have a weak Internet downloading the small files does not work properly. What is the best tuning for CURLOPT_CONNECTTIMEOUT, CURLOPT_LOW_SPEED_LIMIT, and CURLOPT_LOW_SPEED_TIME?

**Comment by bagder** [comment]
> the small files does not work properly

What does this mean?

also, libcurl works exactly the same no matter which network you use it over. libcurl has no knowledge of such details. A slower network just makes things take longer than over faster networks. You might need to take that into account if you use timeout options.

**Reply by khtorabi** [reply]
Hi Daniel. You are right. I have compared with browsers and I have the same result. Could you please give me your idea about the bellow code?
Curl_loop_ptr is equal to zero. but each failing download the amount of this increases by one. After that, I will tune some parameters like this. 
                    if( Curl_loop_ptr ){ 
                        curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT, Curl_loop_ptr ); 
                        ufrea = 120000/( Curl_loop_ptr * Curl_loop_ptr); if( ufrea < 1000 ) ufrea = 1000;
                        curl_easy_setopt(curl, CURLOPT_LOW_SPEED_LIMIT, ufrea ); 
                        ufrea = Curl_loop_ptr * 2; if( ufrea > 60 ) ufrea = 60;
                        curl_easy_setopt(curl, CURLOPT_LOW_SPEED_TIME, ufrea ); 
                    }
